homeassistant/components/shelly/config_flow.py

```diff
@@ -179,12 +198,14 @@ async def async_step_credentials(
                 user_input[CONF_USERNAME] = "admin"
             try:
                 device_info = await validate_input(
-                    self.hass, self.host, self.info, user_input
+                    self.hass, self.host, self.port, self.info, user_input
                 )
             except InvalidAuthError:
                 errors["base"] = "invalid_auth"
             except DeviceConnectionError:
                 errors["base"] = "cannot_connect"
+            except CustomPortNotSupported:
+                errors["base"] = "custom_port_not_supported"
```

Comment:
Please make sure to add a test to cover this

Reply:
Honestly I'm not sure this code path can ever be executed as "gen1" devices are not supported so they cannot be reauth with custom port.

Reply:
If it can't be executed, what is it here for?

Reply:
Removed as I debugged and is not possibile to get there